python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -536,6 +541,70 @@ def _resubmit(self):
         f.close()
         return self._common_submit(new_script_filename)
 
+    def _blockResubmit(self):
+        """Resubmit a DIRAC job that was submitted with bulk submission. This requires writing a new dirac-script for the individual job."""
+        j = self.getJobObject()
+        parametric = False
+
+        if j.master is None:
+            scriptDir = j.getInputWorkspace().getPath()
+        else:
+            scriptDir = j.master.getInputWorkspace().getPath()
+
+        diracScriptFiles = []
+        for fileName in os.listdir(scriptDir):
+            if fnmatch.fnmatch(fileName, 'dirac-script-*.py'):
+                diracScriptFiles.append(fileName)
+
+        for diracScript in diracScriptFiles:
+            script_path = os.path.join(scriptDir, diracScript)
+            # Check old script
+            if not os.path.exists(script_path):
+                raise BackendError('Dirac', 'No "dirac-script.py" found in j.inputdir')
+
+            # Read old script
+            f = open(script_path, 'r')
+            script = f.read()
+            f.close()
+            # Is the subjob we want in there?
+            if not ("sjNo='%s'" % j.fqid) in script:
+                continue
+
+            #First pick out the imports etc at the start
+            newScript =  re.compile(r'%s.*?%s' % ('resultdict = {}',"dirac = Dirac.*?\(\)\n"),re.S).search(script).group(0)
+            newScript += '\n'
+            #Now pick out the job part
+            start = "sjNo='%s'" % j.fqid
+            newScript += re.compile(r'%s.*?%s' % (start,"resultdict.update\({sjNo : result\['Value'\]}\)"),re.S).search(script).group(0)
+            newScript += '\noutput(resultdict)'
+            
+            # Modify the new script with the user settings
+
+            start_user_settings = '# <-- user settings\n'
+            new_script = newScript[
+                :newScript.find(start_user_settings) + len(start_user_settings)]
+
+            job_ident = get_job_ident(newScript.split('\n'))
+            for key, value in self.settings.iteritems():
+                if str(key).startswith('set'):
+                    _key = key[3:]
+                else:
+                    _key = key
+                if type(value) is str:
+                    template = '%s.set%s("%s")\n'
+                else:
+                    template = '%s.set%s(%s)\n'
+                new_script += template % (job_ident, str(_key), str(value))
+            new_script += newScript[newScript.find('# user settings -->'):]
+
+            # Save new script
+            new_script_filename = os.path.join(j.getInputWorkspace().getPath(), 'dirac-script.py')
+            f = open(new_script_filename, 'w')
```

Comment:
again open as a context:
```python
with open(new_script_filename, 'w') as f:
    f.write(new_script)
```

I think then flush should be called automatically. In fact isn't it called automatically already in either the write or close. Do we need a separate flush here?

Reply:
Thanks, good point; I should have updated these when I copied them.